matchzoo/engine/base_model.py

```diff
@@ -45,22 +48,34 @@ def get_default_params(cls) -> engine.ModelParams:
             ...
             ...     @classmethod
             ...     def get_default_params(cls):
-            ...         params = engine.ModelParams()
-            ...         params['num_eggs'] = 512
-            ...         params['ham_type'] = 'Parma Ham'
+            ...         params = engine.ParamTable()
+            ...         params.add(engine.Param('num_eggs', 512))
+            ...         params.add(engine.Param('ham_type', 'Parma Ham'))
             ...         return params
             >>> my_model = MyModel()
             >>> my_model.build()
             512 eggs
             and Parma Ham
 
+        Notice that all parameters must be serialisable for the entire model
+        to be serialisable. Therefore, it's strongly recommended to use python
+        native data types to store parameters.
+
         :return: model parameters
 
         """
-        return engine.ModelParams()
+        params = engine.ParamTable()
+        params.add(engine.Param('name'))
+        params.add(engine.Param('model_class', cls))
+        params.add(engine.Param('input_shapes'))
+        params.add(engine.Param('task'))
+        params.add(engine.Param('metrics'))
+        params.add(engine.Param('loss'))
+        params.add(engine.Param('optimizer'))
```

Comment:
I'm confused by this part, we add Param class instance to param table here without a value (only key)

Reply:
Yes, only key. The value is `None` by default, which means "not filled."